app/src/main/java/cz/muni/fi/pv256/movio2/fk410022/ui/adapter/MovieAdapter.java

```diff
@@ -52,51 +54,52 @@
         }
     }
 
-    public MovieAdapter(Film[] myDataset, OnItemClickListener listener) {
+    public MovieAdapter(List<Film> myDataset, OnFilmClickListener listener, FilmListType type) {
+        this(myDataset.toArray(new Film[myDataset.size()]), listener, type);
+    }
+
+    public MovieAdapter(Film[] myDataset, OnFilmClickListener listener, FilmListType type) {
         mDataset = myDataset;
         mListener = listener;
+        mType = type;
     }
 
     @Override
     public MovieAdapter.ViewHolder onCreateViewHolder(final ViewGroup parent,
                                                       int viewType) {
         View v = LayoutInflater.from(parent.getContext())
                 .inflate(R.layout.view_item, parent, false);
-        Log.i(TAG, "create viev holder:");
         return new ViewHolder(v);
     }
 
     @Override
     public void onBindViewHolder(final ViewHolder holder, int position) {
-        Context context = holder.context;
+        final Context context = holder.context;
         final Film film = mDataset[position];
+        holder.imageView.setImageDrawable(null); // clear
+        holder.detail.setBackgroundColor(Utils.addAlphaToColor(ContextCompat
+                .getColor(context, R.color.palette_default), 0.5f));
```

Comment:
pozadie by mohlo byt trosku menej transparentne, text je tazko citatelny.
keby ste dodrzali povodny design, vyzeralo by to krajsie s backdrop obrazkami roztiahnutymi na sirku :)